BUILD.bazel

```diff
@@ -0,0 +1,151 @@
+    "mocha-patch": "lib/mocha/mocha",
+    "webapis-notification": "lib/browser/webapis-notification",
+    "zone-patch-promise-test": "lib/testing/promise-testing",
+    "zone-patch-resize-observer": "lib/browser/webapis-resize-observer",
+    "webapis-rtc-peer-connection": "lib/browser/webapis-rtc-peer-connection",
+    "zone-patch-rxjs": "lib/rxjs/rxjs",
+    "webapis-shadydom": "lib/browser/shadydom",
+    "zone-patch-socket-io": "lib/extra/socket-io",
+    "zone-patch-user-media": "lib/browser/webapis-user-media",
+    "zone-testing": "lib/testing/zone-testing",
+    "zone-testing-bundle": "lib/browser/rollup-legacy-test-main",
+    "zone-testing-node-bundle": "lib/node/rollup-test-main",
+}
+
+ES2015_BUNDLES = {
+    "zone-evergreen": "lib/browser/rollup-main",
+    "zone-evergreen-testing-bundle": "lib/browser/rollup-test-main",
+}
+
+# the rollup bundles
+rollup(ES5_BUNDLES)
+rollup(ES2015_BUNDLES)
+
+# the es5 filegroups
+[
+    filegroup(
+        name = b + ".es5",
+        srcs = [":" + b],
+        output_group = "es5_umd",
+    )
+    for b in ES5_BUNDLES
+]
+
+# the es5.min filegroups
+[
+    filegroup(
+        name = b + ".es5.min",
+        srcs = [":" + b],
+        output_group = "es5_umd_min",
+    )
+    for b in ES5_BUNDLES
+]
+
+# the es2015 filegroups
+[
+    filegroup(
+        name = b + ".umd",
+        srcs = [":" + b],
+        output_group = "umd",
+    )
+    for b in ES2015_BUNDLES
+]
+
+# the es2015.min filegroups
+[
+    filegroup(
+        name = b + ".umd.min",
+        srcs = [":" + b],
+        output_group = "umd_min",
+    )
+    for b in ES2015_BUNDLES
+]
+
+# Extract and rename each es5 bundle to a .js and .min.js in the dist/ dir
+[
+  genrule(
+    name = b + "-dist",
+    srcs = [b + ".es5", b + ".es5.min"],
+    outs = ["dist/" + b + ".js", "dist/" + b + ".min.js"],
+    cmd = " && ".join([
+      "mkdir -p $(@D)/dist",
+      "mv $(locations //:" + b + ".es5) $(@D)/dist/",
+      "mv $(locations //:" + b + ".es5.min) $(@D)/dist/",
+      "mv $(@D)/dist/" + b + ".es5umd.js $(@D)/dist/" + b + ".js",
+      "mv $(@D)/dist/" + b + ".min.es5umd.js $(@D)/dist/" + b + ".min.js",
+    ])
+  )
+  for b in ES5_BUNDLES
+]
+
+# Extract and rename each es2015 bundle to a .js and .min.js in the dist/ dir
+[
+  genrule(
+    name = b + "-dist",
+    srcs = [b + ".umd", b + ".umd.min"],
+    outs = ["dist/" + b + ".js", "dist/" + b + ".min.js"],
+    cmd = " && ".join([
+      "mkdir -p $(@D)/dist",
+      "mv $(locations //:" + b + ".umd) $(@D)/dist/",
+      "mv $(locations //:" + b + ".umd.min) $(@D)/dist/",
+      "mv $(@D)/dist/" + b + ".umd.js $(@D)/dist/" + b + ".js",
+      "mv $(@D)/dist/" + b + ".min.umd.js $(@D)/dist/" + b + ".min.js",
+    ])
+  )
+  for b in ES2015_BUNDLES
+]
+
+genrule(
+    name = "zone_externs",
+    srcs = ["//lib:zone_externs"],
+    outs = ["dist/zone_externs.js"],
+    cmd = "cp $< $(@D)/zone_externs.js",
+)
+
+npm_package(
```

Comment:
we should add an e2e test for the structure and content of the package so that the `genrule` spaghetti above can be safely refactored out.
something like https://github.com/angular/angular/blob/master/packages/bazel/test/ng_package/core_package.spec.ts maybe?

Reply:
Sure, got it, I have added test cases.